test/support/fake_erlang.ex

```diff
@@ -0,0 +1,5 @@
+defmodule FakeErlang do
+  use TestAgent, %{system_info: 'x86_64-apple-darwin20.2.0'}
```

Comment:
I don't think `FakeErlang` needs to be a test agent. You could implement the method below as `def system_info(:system_architecture), do: 'x86_64-apple-darwin20.2.0'`, and then you don't need the `setup` block where you have to start the process, and you don't need to pass `fake_erlang` to the `test` block either -- putting it in `Application.put_env` alone would do the trick.

Then again, I notice the same would apply to `FakeOS` and `FakeSystem`, which are also used in these tests. There might be a reason why these are `TestAgent`s that I'm missing? (ping @jeffkreeftmeijer: is there?) If there isn't, I think we could turn them all into plain modules -- doesn't have to be in this PR.

Reply:
I have no idea. I'll refer to Jeff for this. I just copied what was there 😬

Reply:
If they don’t have internal state, they don’t have to be agents. That would indeed save us from having to start their processes, because they could just be modules. Most of these are agents because of the idea we might want to update their returned values during the tests, but if we don’t do that, we could make them non-agent modules and return static values.